### PR TITLE
Refactor \DoctrineModule\Form\Element\Proxy

### PR DESCRIPTION
**Pull request by tomphp**
No change in functionality whatsoever here.

I was reading through this code to solve a problem I was having and found the multiple indentation and embedded "if else"s were making it hard to follow.

If this is accepted then I'll happily do more of this, if not then no worries :smile:

### CODE REVIEW

**Comment by gianarb** [comment]
Hi! @tomphp can you rebase this commit please?

**Comment by tomphp** [comment]
Wow! I don't even remember this! Let me take a look and see what I can do.

**Comment by gianarb** [comment]
This PR is too old right know, I just close it.
@tomphp just ping me if you are interested to update it!
